src/clients/asset-client.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const Client = require('./base-client')
+const simpleFetch = require('../utils/simple-fetch')
+const mime = require('mime-types')
+
+class AssetClient {
+  constructor (url) {
+    this.client = new Client({
+      name: 'AssetClient',
+      url: url,
+      contentType: 'application/vnd.api+json'
+    })
```

Comment:
I would use two clients here. One for the assets api and one for the file download / display.